Yoast/wordpress-seo
# Resets the child permalinks

## Context


* When a term has changed we want to reset the permalinks of the children as well, because the permalink might contain the slug of its parent.

## Summary


This PR can be summarized in the following changelog entry:

* Fixes a bug where the indexables table could contain incorrect permalinks. 

## Relevant technical choices:

* Decided to duplicate the code in the test for the indexable builder, we could DRY this later. But it might take some time to get that done.
* Added this piece of code, because in current implementation the `$indexable_before` has a reference to `$indexable` and that means the values are always the same. 
```php
		$indexable_before = $this->indexable_repository
			->query()
			->create( $indexable->as_array() );
```

## Test instructions

This PR can be tested by following these steps:

* Enable breadcrumbs under search appearance and set the category as taxonomy to show in the breadcrumbs.
* Have the permalink structure set to contain `%category%`
* Create a category named `aaa`
* Create a post named `post with aaa` and assign the category `aaa`
* Change the category slug from `aaa` to `bbb`
* Visit the indexables table and see that for the post the permalink / permalink_hash are set to a value that contains the parent slug.
* Create a page named `parent`
* Create another page named `child` and set its parent to `parent`
* Change the permalink for `parent` to something else like. `my-parent`
* Visit the indexables table and see that for the `child` page the permalink / permalink_hash are set to a value that contains the parent slug.

## UI changes

## Documentation

## Quality assurance


Fixes #
